File: MQ/Python/receive.py
from azure.eventhub import EventHubConsumerClient
import os
from dotenv import load_dotenv
load_dotenv()

CONSUMER_GROUP = os.getenv("CONSUMER_GROUP")
EVENT_HUB_COMPATIBLE_ENDPOINT = os.getenv("EVENT_HUB_COMPATIBLE_ENDPOINT")

def on_event_batch(partition_context, events):
    for event in events:
        print(f"Received: {event.body_as_str()}")
    partition_context.update_checkpoint()

def on_error(partition_context, error):
    if partition_context:
        print("An exception: {} occurred during receiving from Partition: {}.".format(
            error,
            partition_context.partition_id
        ))
    else:
        print("An exception: {} occurred during the load balance process.".format(error))

def main():
    client = EventHubConsumerClient.from_connection_string(
        conn_str=EVENT_HUB_COMPATIBLE_ENDPOINT,
        consumer_group=CONSUMER_GROUP,
    )
    try:
        with client:
            client.receive_batch(
                on_event_batch=on_event_batch,
                on_error=on_error
            )
    except KeyboardInterrupt:
        print("Receiving has stopped.")

if __name__ == '__main__':
    main()

File: MQ/Nodejs/receive.js
const { EventHubConsumerClient } = require('@azure/event-hubs');

require("dotenv").config();

const connectionString = process.env["EVENTHUB_CONNECTION_STRING"] || "";
const eventHubName = process.env["EVENTHUB_NAME"] || "";
const consumerGroup = process.env["CONSUMER_GROUP_NAME"] || "";

const consumerClient = new EventHubConsumerClient(consumerGroup,
  connectionString, eventHubName);
this.subscription = consumerClient.subscribe({
  processEvents: async (events, context) => {
    console.log(`events: ${JSON.stringify(events)}`);
    await context.updateCheckpoint(events[events.length - 1]);
  },
  // processError: async (err, context) => {
  //   console.log(`Error: ${err}, context: ${context}`);
  // },
});